From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MrPowerGamerBR <git@mrpowergamerbr.com>
Date: Wed, 5 Jun 2024 15:20:00 -0300
Subject: [PATCH] SparklyPaper: Reset dirty flag when loading maps from the
 disk

Original project: https://github.com/SparklyPower/SparklyPaper

By default, the server will start rewriting all map datas to the disk after loading it, even if the map didn't have any changes

This also slows down world saving a lot if you have a lot of maps

diff --git a/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java b/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
index dbad8927a43c4bffcfe4e7596c50ab4c7cc9d7d9..d96051f07cb6a9327f99c29c62b62919336af237 100644
--- a/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
+++ b/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
@@ -156,6 +156,7 @@ public class MapItemSavedData extends SavedData {
         }
 
         this.vanillaRender.buffer = colors.array(); // Paper - Use Vanilla map renderer when possible
+        this.setDirty(false); // SparklyPaper - reset dirty flag when loading maps from the disk (context for updates: this modification is at the end of the map "load" function)
     }
 
     public static MapItemSavedData createFresh(
